### DISCUSSION TITLE
ChakraProvider needed in each file?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2777] [Q&A] [answered]
The User Guide states that ChakraProvider should be in index.js for React apps.

But, the @chakra-ui template places it in App.js (and not in index.js.

Does it need to be in both files? Every file that uses Chakra-UI?

Or just at a top level? And if so, why does the template put it in a different file than the User Guide?

**Comment by chr-ge** [comment, marked as answer]
You only need to wrap your application at the top level in either the `index.js` file or your `App.js` file.
It's up to you in which file you choose to do it.

**Reply by with-heart** [reply]
@zmagickap I see what you mean and we should probably remove the references to `index.js` and `index.tsx` as `ChakraProvider` doesn't _have_ to be rendered there. Like @chr-ge said, it's really up to you how you structure your app, as long as `ChakraProvider` wraps the content of your app (meaning anything built using Chakra components).
